File: backend/app/schemas/indexing.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class TriggerBranchIndexingRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    commit_hashed: str | None = None


class IndexingJobResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    workspace_id: int
    branch_id: int
    status: str
    progress_pct: int
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime

File: frontend/src/api/types/indexing.ts
/**
 * `status` is a free string on the backend (not formally tied to the
 * BranchIndexingStatus enum) — don't assume it only ever holds those 5 values.
 */
export interface IndexingJobResponse {
  id: number;
  workspace_id: number;
  branch_id: number;
  status: string;
  progress_pct: number;
  error_message?: string | null;
  created_at: string;
  updated_at: string;
}
